### ISSUE TITLE
Warn/error when a repo can't be updated

### ISSUE THREAD

**Issue by gerkey** [vcstools/wstool#102]
I had a situation today in which one of the git repos in my workspace didn't have an upstream branch configured (how I got into that situation is another question). I used `wstool update` on the workspace, and from its error- and warning-free output, I thought that all of my repos had been updated. But it turned out, after much digging around looking for the cause of an inexplicable problem, that that repo was not being updated.

I can understand that `wstool update` might not be wiling or able to update a repo when `git pull` doesn't work (as in this case because there's no upstream branch configured), but it would be great to get a warning or error to let me know that something went wrong.

**Comment by wjwwood** [comment]
A minimal reproducible example would be useful here. The issue probably lies within the `vcstools` library. I don't have time to dig into it right now, but with a failing example someone else might have time to do so.

**Comment by gerkey** [comment]
I spent some time trying to put together an example, but I failed. Now I'm seeing the behavior that you (@wjwwood) had previously suggested would be the case, which is for `wstool up` to switch a repo back to the branch specified in the `.rosinstall` configuration and update there. Previously I was in a state where I had a repo on a branch different from that specified in the `.rosinstall` and without an upstream branch set, and `wstool up` wasn't doing anything, though it appeared to.

I'll close this for failure to reproduce.
